src/test/java/com/rocketden/main/api/RoomTests.java

```diff
@@ -315,4 +313,163 @@ public void createAndJoinRoomNoUser() throws Exception {
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(createRequest)))
+                .andDo(print()).andExpect(status().isCreated())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        RoomDto room = Utility.toObject(jsonResponse, RoomDto.class);
+
+        UserDto user = new UserDto();
+        user.setNickname("User");
+
+        JoinRoomRequest joinRequest = new JoinRoomRequest();
+        joinRequest.setRoomId(room.getRoomId());
+        joinRequest.setUser(user);
+
+        result = this.mockMvc.perform(put(PUT_ROOM)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(joinRequest)))
+                .andDo(print()).andExpect(status().isOk())
+                .andReturn();
+
+        jsonResponse = result.getResponse().getContentAsString();
+        room = Utility.toObject(jsonResponse, RoomDto.class);
+
+        // Verify that we have created a room with 2 users
+        assertEquals(2, room.getUsers().size());
+
+        // Non-host tries to change hosts
+        UpdateHostRequest updateHostRequest = new UpdateHostRequest();
+        updateHostRequest.setInitiator(user);
+        updateHostRequest.setNewHost(host);
+
+        ApiError ERROR = RoomError.INVALID_PERMISSIONS;
+
+        result = this.mockMvc.perform(put(String.format(PUT_ROOM_HOST, room.getRoomId()))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(updateHostRequest)))
+                .andDo(print()).andExpect(status().is(ERROR.getStatus().value()))
+                .andReturn();
+
+        jsonResponse = result.getResponse().getContentAsString();
+        ApiErrorResponse actual = Utility.toObject(jsonResponse, ApiErrorResponse.class);
+
+        assertEquals(ERROR.getResponse(), actual);
+
+        // Room does not exist
+        updateHostRequest = new UpdateHostRequest();
+        updateHostRequest.setInitiator(host);
+        updateHostRequest.setNewHost(user);
+
+        ERROR = RoomError.NOT_FOUND;
+
+        result = this.mockMvc.perform(put(String.format(PUT_ROOM_HOST, "999999"))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(updateHostRequest)))
+                .andDo(print()).andExpect(status().is(ERROR.getStatus().value()))
+                .andReturn();
+
+        jsonResponse = result.getResponse().getContentAsString();
+        actual = Utility.toObject(jsonResponse, ApiErrorResponse.class);
+
+        assertEquals(ERROR.getResponse(), actual);
+
+        // New host does not exist in the room
+        updateHostRequest = new UpdateHostRequest();
+        updateHostRequest.setInitiator(host);
+        updateHostRequest.setNewHost(new UserDto());
```

Comment:
If I understand correctly, you are testing multiple different errors in the same method. Thoughts on taking these out and separating them into separate methods?

If you are concerned about the large amount of duplicate code beforehand, maybe place that in a setup helper function, and have a separate test (you may already) to ensure that that code is indeed working. Let me know your thoughts.

Reply:
Yeah that's a good idea, done!